pipeline/load.py

```diff
@@ -0,0 +1,220 @@
+    "Fetches all alert values from RDS"
+    return fetch_all(cursor, "alerts", "alert_value", "alert_id")
+
+
+def get_all_networks(cursor: cursor) -> dict:
+    "Fetches all network values from RDS"
+    return fetch_all(cursor, "networks", "network_name", NETWORK_ID)
+
+
+def get_all_statuses(cursor: cursor) -> dict:
+    "Fetches all statuses from RDS"
+    return fetch_all(cursor, "statuses", STATUS, "status_id")
+
+
+def get_all_magtypes(cursor: cursor) -> dict:
+    "Fetches all magtypes values from RDS"
+    return fetch_all(cursor, "magtypes", "magtype_value", MAGTYPE_ID)
+
+
+def get_all_types(cursor: cursor) -> dict:
+    "Fetches all earthquake types from RDS"
+    return fetch_all(cursor, "types", "type_value", TYPE_ID)
+
+
+def get_or_add_id(value: str, all_items: dict, add_to_db_function) -> int:
+    """Retrieves the ID for a given value, adding it to the database if not present"""
+    logging.info(f"Retrieving ID for {value}")
+    if value in all_items:
+        return all_items[value]
+    logging.info(f"{value} doesn't exist in database - inserting it now")
+    new_id = add_to_db_function(value)
+    if new_id is not None:
+        all_items[value] = new_id
+    return new_id
+
+
+def add_network_to_db(conn: connection, cursor: cursor, network_value: str) -> int:
+    """Adds the network value to the RDS"""
+    query = """INSERT INTO networks (network_name) VALUES (%s) RETURNING network_id"""
+    return execute_insert(conn, cursor, query, (network_value,), NETWORK_ID)
+
+
+def get_network_id(conn: connection, cursor: cursor, network_value: str, all_networks: dict) -> int:
+    """Gets the network_id for the provided network"""
+    return get_or_add_id(network_value, all_networks, lambda value_to_add: add_network_to_db(conn, cursor, value_to_add))
+
+
+def add_magtype_to_db(conn: connection, cursor: cursor, magtype_value: str) -> int:
+    """Adds the magType value to the RDS"""
+    query = """INSERT INTO magtypes (magtype_value) VALUES (%s) RETURNING magtype_id"""
+    return execute_insert(conn, cursor, query, (magtype_value,), MAGTYPE_ID)
+
+
+def get_magtype_id(conn: connection, cursor: cursor, magtype_value: str, all_magtypes: dict) -> int:
+    """Gets the magtype_id for the provided magType value"""
+    return get_or_add_id(magtype_value, all_magtypes, lambda value_to_add: add_magtype_to_db(conn, cursor, value_to_add))
+
+
+def add_type_to_db(conn: connection, cursor: cursor, earthquake_type: str) -> int:
+    """Adds the earthquake type to the RDS"""
+    query = """INSERT INTO types (type_value) VALUES (%s) RETURNING type_id"""
+    return execute_insert(conn, cursor, query, (earthquake_type,), TYPE_ID)
+
+
+def get_type_id(conn: connection, cursor: cursor, earthquake_type: str, all_types: dict) -> int:
+    """Gets the type_id for the provided earthquake type value"""
+    return get_or_add_id(earthquake_type, all_types, lambda value_to_add: add_type_to_db(conn, cursor, value_to_add))
```

Comment:
same as above, these functions seem quite repetitive. Any way to pass the column you want to access/write to as a parameter instead

Reply:
Same as the above, I don't think it's worse the hassle since each has a different query and passed in lambda